financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/auth/extensions/request/validator/FSRequestObjectValidationExtension.java

```diff
@@ -155,4 +166,155 @@ protected boolean isRegulatory(OAuth2Parameters oAuth2Parameters) throws Request
+            String sanitizedMessage = response.getViolationMessage().replaceAll("[\r\n]+", " ");
+            log.error("Request object validation failed: " + sanitizedMessage);
+            throw new RequestObjectException(RequestObjectException.ERROR_CODE_INVALID_REQUEST, sanitizedMessage);
+        }
+    }
+
+    /**
+     * Performs accelerator layer default validations.
+     *
+     * @param fsRequestObject The request object to be validated.
+     * @param dataMap         Additional data required for validation, such as scope-related information.
+     * @return A ValidationResponse indicating whether the request is valid.
+     */
+    private ValidationResponse defaultValidateRequestObject(FSRequestObject fsRequestObject,
+                                                            Map<String, Object> dataMap) {
+
+        String fsRequestObjectViolation = FinancialServicesValidator.getInstance().getFirstViolation(fsRequestObject);
+
+        if (StringUtils.isEmpty(fsRequestObjectViolation)) {
+            try {
+                if (!isClientIdAndScopePresent(fsRequestObject)) {
+                    return new ValidationResponse(false, "Client Id and scope are mandatory to" +
+                            " include in the request object.");
+                }
+                String scopeViolation = validateScope(fsRequestObject, dataMap);
+                if (StringUtils.isEmpty(scopeViolation)) {
+                    return new ValidationResponse(true);
+                } else {
+                    return new ValidationResponse(false, scopeViolation);
+                }
+            } catch (RequestObjectException e) {
+                return new ValidationResponse(false, e.getMessage());
+            }
+        } else {
+            return new ValidationResponse(false, fsRequestObjectViolation);
+        }
+    }
+
+    private boolean isClientIdAndScopePresent(FSRequestObject fsRequestObject) throws RequestObjectException {
+        JSONObject jsonObject = fsRequestObject.getSignedJWT().getPayload().toJSONObject();
+        final String clientId = jsonObject.getAsString(IdentityCommonConstants.CLIENT_ID);
+        String scope = jsonObject.getAsString(IdentityCommonConstants.SCOPE);
+        if (StringUtils.isBlank(clientId) || StringUtils.isBlank(scope)) {
+            log.error("Client id or scope cannot be empty");
+            throw new RequestObjectException("Client id or scope cannot be empty");
+        }
+        return true;
+    }
+
+    /**
+     * Validates scope.
+     *
+     * @param fsRequestObject
+     * @param dataMap
+     * @return
+     */
+    private String validateScope(FSRequestObject fsRequestObject, Map<String, Object> dataMap) {
+
+        try {
+            //remove scope claim
+            JWTClaimsSet claimsSet = fsRequestObject.getClaimsSet();
+            JSONObject claimsSetJsonObject = claimsSet.toJSONObject();
+            if (claimsSetJsonObject.containsKey("scope")) {
```

Comment:
Let's move the strings used in multiple places to constants.

Reply:
Fixed with 69b55aa0ba7b9206573f897b8ab47eaa78466fbe